### DISCUSSION TITLE
/finances/v0/financialEventGroups/{eventGroupId}/financialEvents InvalidInput

### DISCUSSION THREAD

**Discussion by houyachao** [amzn/selling-partner-api-models#587] [Ideas]
Request/finances/where v0 / financialEventGroups / {eventGroupId} / financialEvents page data accessed by the NextToken, return to {" errors ": [{"code": "InvalidInput", "message": "$errorResponse.Message", "details": ""}]}. How to fix it?

**Comment by RobertAbaev** [comment]
Same problem here with `GET /finances/v0/financialEvents`. For me it only happens with NA sellers, all requests that provide `NextToken` fail with:
```json
{
    "errors": [
        {
            "code": "InvalidInput",
            "message": "Invalid Input",
            "details": ""
        }
    ]
}
```
And it is not a response size problem (as mentioned [here ](https://developer-docs.amazon.com/sp-api/docs/finances-api-reference#get-financesv0financialevents) under `MaxResultsPerPage` description) since I can retrieve same data just by doing several requests with smaller time window (`PostedAfter` & `PostedBefore`) so they won't generate `NextToken` for me.

**Reply by shlomieconnect** [reply]
did you Url encode the NextToken?

**Reply by RobertAbaev** [reply]
@shlomieconnect ~~just tried it, looks like it's working, thank you! 
It's just weird that previously we had our pipelines running for over a year without encoding and we have never encountered this problem. Did they change `NextToken` generation without announcing or did I miss something?~~ 
No, scratch that. Looks like it was a one-off. Still getting error most of the time, very inconsistent behavior.

**Reply by estebanangel** [reply]
same here, this started to pop up yesterday. no clue what's going on. hope to hear something from Amazon!

**Reply by saeedasma** [reply]
I am having the same issue. It started happening on 07/26. Did you find a solution for it? Please share if you find anything.

**Reply by shlomieconnect** [reply]
I just wanted to clarify. There are 2 types of errors that are being conflated here.
1. code: "Invalid Input", message: "Invalid Input"
2. code: "Invalid Input", message: "$errorResponse.Message".

Both of these issues are old issues. 

Error 1, is documented, there are 2 reasons it can happen, either the request is malformed, or the response is larger than 10 MB. If the original request works, and adding a NextToken **always** fails, no matter the group, most probably the NextToken is malformed, hence my suggestion to UrlEncode the NextToken. If the NextToken only fails **some groups**, then most probably the response is larger then 10 MB. In this case, try to add a MaxResultsPerPage parameter, and lowering the value until it succeeds.

Error 2, is undocumented, in the past this error has happened, however, it has happened only sporadically. 

The new issue is as follows, Error 2 has spiked since July 26, it is now happening in a large percentage of requests. Whereas until now a simple retry policy worked, now, it is almost untenable. I have retried the same request more than 10 times, only to have it work at a later retry. It's not feasible to retry a request 10+ times hoping that the 11th retry will work. 

I have not seen a spike in error 1, however this doesn't mean it doesn't exist, I myself have not experienced it.

**Reply by jimmy-ross** [reply]
the explanation of Error 2 reflects what we are seeing too, we have a lot of retry mechanisms in place but it is currently untenable since it started spiking.

**Reply by RobertAbaev** [reply]
Just tried doing what @houyachao suggested below (mindless reattempt with the same token) and it does work after a few attempts. We have reported the issue to SP-API support, waiting for an answer.

**Comment by saeedasma** [comment]
I am using the C# client library to call ListFinancialEvents. It gives the same error:
{
    "errors": [
        {
            "code": "InvalidInput",
            "message": "Invalid Input",
            "details": ""
        }
    ]
}

I also tried using the C# Sdk and that is returning this error:
'Error converting value 26 to type 'IO.Swagger.Model.BigDecimal'. Path 'payload.FinancialEvents.ShipmentEventList[0].ShipmentItemList[0].ItemChargeList[0].ChargeAmount.CurrencyAmount', line 49, position 189.'

**Reply by estebanangel** [reply]
Yes, same here using Python, the problem happens with the NextToken parameter, it's a bug from their side, this has been stable for 2 years in our case and started to fail 2 days ago. we need their attention here!

**Comment by jimmy-ross** [comment]
Hi! I created an issue here too, this seems to be a major bug on Amazons side as we also have been using the Finance API for over 2 years: https://github.com/xapix-io/seller-reporting-sync/issues/1834

I created a case with Amazon and they are investigating, please do so on your sides too so the urgency is made clear.

**Comment by houyachao** [comment]
We found a solution on our side, if Amazon returns code=400, we are while keep retrying this request until code=200. It seems to solve the problem for now.

**Comment by danielYaakov** [comment]
Same issue here 😥.
We have a lot of retries mechanism but in most cases is not enough!
